Fixes several build warnings.

diegoreymendez opened the pull request:
This PR addresses several build warnings in Xcode 13.

I've tried only including simple warnings so that the review is easy.

## To test:

## Regression Notes

1. Potential unintended areas of impact

I don't think there's space for issues at runtime, and I believe any issues with my changes should only affect the code at build-time, meaning that if the code compiles I trust everything should be fine.

It's important though that the reviewer makes sure my changes are logically correct.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

diegoreymendez committed:
Fixes several build warnings.

jkmassel commented on WordPress/Classes/ViewRelated/Gutenberg/GutenbergImageLoader.swift:
```diff
@@ -13,11 +13,11 @@ class GutenbergImageLoader: NSObject, RCTImageURLLoader {
-    func canLoadImageURL(_ requestURL: URL!) -> Bool {
-        return !requestURL.isFileURL
+    func canLoadImageURL(_ requestURL: URL) -> Bool {
+        return requestURL.isFileURL
```
This seems to change how this method works, not just its signature – do we want that?

mokagio replied:
+1 on this question.

The docs for `isFileURL` say: Returns true if the scheme is `file:`.

diegoreymendez replied:
Sorry this was a mistake after seeing so many IUOs.  I'm glad I had your eyes on this review!

diegoreymendez replied:
Rolled back this change.

jkmassel commented on WordPress/Classes/ViewRelated/Post/PostSettingsViewController.m:
```diff
@@ -459,7 +459,6 @@ - (CGFloat)tableView:(UITableView *)tableView heightForFooterInSection:(NSIntege
 
 - (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath
 {
-    CGFloat width = CGRectGetWidth(self.tableView.frame);
```
Just to confirm – this is just an unused variable?

diegoreymendez replied:
Yes.

jkmassel commented on WordPress/Classes/ViewRelated/Themes/ThemeBrowserCell.swift:
```diff
@@ -209,7 +209,7 @@ open class ThemeBrowserCell: UICollectionViewCell {
 
         if let screenshotWidth = presenter?.screenshotWidth {
-            queryItems.append(URLQueryItem(name: "w", value: "\(presenter!.screenshotWidth)"))
+            queryItems.append(URLQueryItem(name: "w", value: "\(screenshotWidth)"))
```
Did swift notice that `presenter` can't be `nil` here? If so, that's super cool!

mokagio replied:
Nope 😢 

![image](https://user-images.githubusercontent.com/1218433/134425241-e77ab137-b137-44f2-93fb-4200a607a939.png)

diegoreymendez replied:
It took me a while to understand the question. 😄 

The reason I noticed we could just use `screenshotWidth` directly is because of the alert that @mokagio shows in his screenshot.

jkmassel commented on WordPress/Classes/ViewRelated/Gutenberg/GutenbergImageLoader.swift:
```diff
@@ -13,11 +13,11 @@ class GutenbergImageLoader: NSObject, RCTImageURLLoader {
     }
 
-    func loadImage(for imageURL: URL!, size: CGSize, scale: CGFloat, resizeMode: RCTResizeMode, progressHandler: RCTImageLoaderProgressBlock!, partialLoadHandler: RCTImageLoaderPartialLoadBlock!, completionHandler: RCTImageLoaderCompletionBlock!) -> RCTImageLoaderCancellationBlock! {
+    func loadImage(for imageURL: URL, size: CGSize, scale: CGFloat, resizeMode: RCTResizeMode, progressHandler: RCTImageLoaderProgressBlock, partialLoadHandler: RCTImageLoaderPartialLoadBlock, completionHandler: @escaping RCTImageLoaderCompletionBlock) -> RCTImageLoaderCancellationBlock? {
```
Should the return type be optional? AFAICT all of the paths return a block?

diegoreymendez replied:
This method is implemented in `GutenbergImageLoader` to comply with protocol `RCTImageURLLoader`, which defines the method like this (with a `nullable` return type):

```objective-c
- (nullable RCTImageLoaderCancellationBlock)loadImageForURL:(NSURL *)imageURL
                                              size:(CGSize)size
                                             scale:(CGFloat)scale
                                        resizeMode:(RCTResizeMode)resizeMode
                                   progressHandler:(RCTImageLoaderProgressBlock)progressHandler
                                partialLoadHandler:(RCTImageLoaderPartialLoadBlock)partialLoadHandler
                                 completionHandler:(RCTImageLoaderCompletionBlock)completionHandler;
```

EDIT: as a side note I tried making this non-optional and the compiler gives me a warning that it doesn't match the return type optionality of the protocol.

jkmassel reviewed (approved):
Just reviewed in the GitHub UI, but I left a few questions

mokagio reviewed (commented):
Thanks @diegoreymendez for this! This is not the first "address warnings" PR from you that I have the pleasure of reviewing. Thank you for looking after the codebase.

diegoreymendez committed:
Rolls back a change made by mistake.

diegoreymendez commented:
Thanks @jkmassel and @mokagio !

Since I addressed your comments rolling back 1 change, and with other comments, could I ask you to revalidate that everything looks fine now?

Thanks!

mokagio commented:
Thanks @diegoreymendez. Looking great. Merged it so everyone branching out or rebasing on top of `develop` can immediately enjoy these changes.

diegoreymendez commented:
Thanks for pushing it forward @mokagio !
